cms/lib/xblock/tagging/tagging.py

```diff
@@ -71,25 +79,36 @@ def save_tags(self, request=None, suffix=None):  # pylint: disable=unused-argume
         """
         Handler to save choosen tags with connected XBlock
         """
-        found = False
-        if 'tag' not in request.params:
-            return Response("The required parameter 'tag' is not passed", status=400)
+        try:
+            posted_data = json.loads(request.body_file.read())
```

Comment:
@cpennington yes, I know about this way. But in the current place we can't use it because in this case we are faced with such exception: `RawPostDataException: You cannot access body after reading from request's data stream`

```
2017-02-01 18:13:52,032 ERROR 26678 [django.request] base.py:256 - Internal Server Error: /preview/xblock/aside-usage-v2:block-v1$:MXX+CS102+2016_3+type@problem+block@fe90caf067104d92a9dbd1ba375a0387::tagging_aside/handler/save_tags
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform-credo/cms/djangoapps/contentstore/views/preview.py", line 75, in preview_handler
    resp = instance.handle(handler, req, suffix)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock/mixins.py", line 86, in handle
    return self.runtime.handle(self, handler_name, request, suffix)
  File "/edx/app/edxapp/edx-platform-credo/common/lib/xmodule/xmodule/x_module.py", line 1339, in handle
    return super(MetricsMixin, self).handle(block, handler_name, request, suffix=suffix)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock/runtime.py", line 1025, in handle
    results = handler(request, suffix)
  File "/edx/app/edxapp/edx-platform-credo/cms/lib/xblock/tagging/tagging.py", line 84, in save_tags
    print '>>>>>', request.json
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/webob/request.py", line 724, in _json_body__get
    return json.loads(self.body.decode(self.charset))
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/lazy/lazy.py", line 28, in __get__
    value = self.__func(inst)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/xblock/django/request.py", line 149, in body
    return self._request.body
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/http/request.py", line 232, in body
    raise RawPostDataException("You cannot access body after reading from request's data stream")
RawPostDataException: You cannot access body after reading from request's data stream
```